myapp/scheduler/stock_api_update.py

```diff
@@ -24,3 +32,26 @@ def stock_api_update():
             index += 1
     except KeyError as e:
         pass
+
+
+def update_existing_stocks():
+    s = WatchStock.objects.all().values("stock_id").distinct()
+    for w in s:
+        stock = Stock.objects.all().filter(symbol=w.get('stock_id'))[0]
```

Comment:
maybe you can optimize this by calling the stock via watched-stock, like this `w.stock`